src/execution/operator/aggregate/physical_window.cpp

```diff
@@ -457,6 +458,98 @@ static bool WindowNeedsRank(BoundWindowExpression *wexpr) {
 	       wexpr->type == ExpressionType::WINDOW_RANK_DENSE || wexpr->type == ExpressionType::WINDOW_CUME_DIST;
 }
 
+template <typename SRC, typename DST>
+static DST GetTypedCell(Vector &source, idx_t index) {
+	switch (source.GetVectorType()) {
+	case VectorType::CONSTANT_VECTOR: {
+		const auto data = ConstantVector::GetData<DST>(source);
+		return data[0];
+	}
+	case VectorType::FLAT_VECTOR:
+		break;
+		// dictionary: apply dictionary and forward to child
+	case VectorType::DICTIONARY_VECTOR: {
+		auto &sel_vector = DictionaryVector::SelVector(source);
+		auto &child = DictionaryVector::Child(source);
+		return GetTypedCell<SRC, DST>(child, sel_vector.get_index(index));
+	}
+	case VectorType::SEQUENCE_VECTOR: {
+		int64_t start, increment;
+		SequenceVector::GetSequence(source, start, increment);
+		return Cast::Operation<int64_t, DST>(start + increment * index);
+	}
+	default:
+		throw InternalException("Unimplemented vector type for GetTypedCell");
+	}
+
+	const auto data = FlatVector::GetData<SRC>(source);
+	return Cast::Operation<SRC, DST>(data[index]);
+}
+
+template <typename SRC, typename DST>
+static inline DST GetDecimalCell(Vector &source, idx_t source_offset, uint8_t width, uint8_t scale) {
+	return CastFromDecimal::Operation<SRC, DST>(GetTypedCell<SRC, SRC>(source, source_offset), width, scale);
+}
+
+template <typename T>
+static T GetCell(ChunkCollection &collection, idx_t column, idx_t index) {
```

Comment:
The GetCell method only seems to be used for window boundaries. Should we perhaps just enforce that the window boundary is always int64 through (optionally) adding a cast on the start_expr and end_expr in the binding phase? That seems like it would simplify this logic here immensely.

Reply:
Yes I was wondering about that when I did the default expression change, but I thought I'd wait for the review. I can do that now or put it in the next one (which is `nth_value`).

Reply:
@ankoh was of the opinion that this could be counterproductive?

Reply:
What we settled on was that the binding would be a Good Idea if we did payload/boundary materialisation one block at a time and only used collections for the input. I think that is where we are headed now, but it should probably be in a different PR.

Reply:
Hm a bit lost now - how does that relate to the logical type of the boundary elements? As far as I can tell the boundary is either a constant value or an expression, in both cases we could simply push a cast to BIGINT which would simplify the code here as we would no longer need to switch on the type. Am I missing something perhaps?

Reply:
I think the concern was just that right now we materialise such calculations for the entire collection - which can be large. But it is a better solution long term, so maybe that is a better than later removing what I did.

Reply:
I think whether or not we materialize the boundaries entirely or in chunks is rather orthogonal to the types of the boundaries, no? I’m mainly proposing to always have boundaries be int64 for simplicity so all the casts and type switching in GetCell are no longer required.